packages/headless/src/features/folding/folding-actions.ts

```diff
@@ -1,14 +1,35 @@
 import {NumberValue, SchemaDefinition, StringValue} from '@coveo/bueno';
-import {createAction} from '@reduxjs/toolkit';
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {
+  AsyncThunkSearchOptions,
+  isErrorResponse,
+} from '../../api/search/search-api-client';
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {Result} from '../../api/search/search/result';
+import {ConfigurationSection, FoldingSection} from '../../state/state-sections';
 import {validatePayload} from '../../utils/validate-payload';
 
+export const collectionMoreResultsAvailableBuffer = 1;
+
+export type CollectionId = string;
+
 export interface RegisterFoldingPayload {
   collectionField?: string;
   parentField?: string;
   childField?: string;
   numberOfFoldedResults?: number;
 }
 
+export interface LoadAllFulfilledReturn {
```

Comment:
In the same theme as the action name, I thin these interface names would be clearer if called:

- `LoadCollectionFulfilledReturn`
- `LoadCollectionRejectedReturn`